superset-frontend/packages/superset-ui-core/src/time-format/formatters/finestTemporalGrain.ts

```diff
@@ -48,26 +48,30 @@ export default function finestTemporalGrain(
   } = useLocalTime ? localTimeUtils : utcUtils;
 
   let formatFunc = formatYear;
-  values.forEach((value: any) => {
-    if (formatFunc === formatYear && isNotFirstMonth(value)) {
-      formatFunc = formatMonth;
-    }
-    if (formatFunc === formatMonth && isNotFirstDayOfMonth(value)) {
-      formatFunc = formatDay;
-    }
-    if (formatFunc === formatDay && hasHour(value)) {
-      formatFunc = formatHour;
-    }
-    if (formatFunc === formatHour && hasMinute(value)) {
-      formatFunc = formatMinute;
-    }
-    if (formatFunc === formatMinute && hasSecond(value)) {
-      formatFunc = formatSecond;
-    }
-    if (formatFunc === formatSecond && hasMillisecond(value)) {
-      formatFunc = formatMillisecond;
-    }
-  });
+  try {
+    values.forEach((value: any) => {
```

Comment:
Why do we need a try/catch to assign `formatFunc`?

Reply:
revert the try-catch block and add the bigint type check to clarify the issue (Cannot convert a BigInt value to a number in Date function)